mmcls/models/backbones/efficientnet_v2.py

```diff
@@ -0,0 +1,395 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import math
+from functools import partial
+from typing import Callable, List, Optional, Tuple
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.cnn.bricks import DropPath
+from mmengine.model import Sequential
+from torch import Tensor
+
+from mmcls.models.backbones.base_backbone import BaseBackbone
+from mmcls.registry import MODELS
+
+
+# Copy from timm
+# Calculate asymmetric TensorFlow-like 'SAME' padding for a convolution
+def get_same_padding(x: int, k: int, s: int, d: int):
+    return max((math.ceil(x / s) - 1) * s + (k - 1) * d + 1 - x, 0)
+
+
+# Dynamically pad input x with 'SAME' padding for conv with specified args
+def pad_same(x,
+             k: List[int],
+             s: List[int],
+             d: List[int] = (1, 1),
+             value: float = 0):
+    ih, iw = x.size()[-2:]
+    pad_h, pad_w = get_same_padding(ih, k[0], s[0], d[0]), get_same_padding(
+        iw, k[1], s[1], d[1])
+    if pad_h > 0 or pad_w > 0:
+        x = F.pad(
+            x,
+            [pad_w // 2, pad_w - pad_w // 2, pad_h // 2, pad_h - pad_h // 2],
+            value=value)
+    return x
+
+
+def conv2d_same(x,
+                weight: torch.Tensor,
+                bias: Optional[torch.Tensor] = None,
+                stride: Tuple[int, int] = (1, 1),
+                padding: Tuple[int, int] = (0, 0),
+                dilation: Tuple[int, int] = (1, 1),
+                groups: int = 1):
+    x = pad_same(x, weight.shape[-2:], stride, dilation)
+    return F.conv2d(x, weight, bias, stride, (0, 0), dilation, groups)
+
+
+class Conv2dSame(nn.Conv2d):
```

Comment:
can it be replaced with `mmcv.Conv2dAdaptivePadding`? https://github.com/open-mmlab/mmcv/blob/fb39e1e568c49fcd97f9a56cb70a89d6693738c7/mmcv/cnn/bricks/conv2d_adaptive_padding.py#L12

Reply:
yes, i will change all my code to mmcls style, not difficult but need some time